src/app/static/src/app/components/projects/ShareProject.vue

```diff
@@ -109,33 +112,48 @@
         },
         data() {
             return {
-                emailsToShareWith: [{value: "", valid: null}],
+                emailsToShareWith: [{value: "", valid: null, validationMessage: "blank"}],
                 open: false,
                 showValidationMessage: false
             }
         },
         methods: {
             cloneProject: mapActionByName("projects", "cloneProject"),
             userExists: mapActionByName("projects", "userExists"),
-            addEmail(e: EmailToShareWith, index: number) {
-                if (e.value) {
-                    if (index == this.emailsToShareWith.length - 1) {
-                        // if blur event fires on the last input
-                        // add another input below it
-                        this.emailsToShareWith.push({
-                            value: "",
-                            valid: null
-                        });
-                    }
-                    this.userExists(e.value)
-                        .then((result: boolean) => {
-                            this.emailsToShareWith[index].valid = result;
+            addEmail() {
+                this.emailsToShareWith.map((email: EmailToShareWith, index: number) => {
```

Comment:
Because the validation of duplicate emails is dependent on changes to other emails, validation must be run over the entire list every time the blur event fires

Reply:
Huh, weirdly didn't include my review comment.. 
This works, but think readability could be improved. Also don't need to consider adding control in the map function, if reinstate method params. Something like this? 
```addEmail(e: EmailToShareWith, index: number) {
                if (e.value && index == this.emailsToShareWith.length - 1) {
                    this.emailsToShareWith.push({
                        value: "",
                        valid: null,
                        validationMessage: "blank"
                    });
                }
                this.emailsToShareWith.map(async (email: EmailToShareWith, index: number) => {
                    if (email.value) {
                        let invalidMsg = null;

                        if (email.value == currentUser) {
                            invalidMsg = "projectsNoSelfShare";
                        } else if (this.emailsToShareWith.filter(val => val.value === email.value).length > 1) {
                            invalidMsg = "duplicateEmails";
                        } else {
                            await this.userExists(email.value)
                                .then((result: boolean) => {
                                    if (!result) {
                                        invalidMsg = "emailNotRegistered"
                                    }}
                                );
                        }

                        this.emailsToShareWith[index].validationMessage = invalidMsg || "";
                        this.emailsToShareWith[index].valid = invalidMsg === null;

                    } else {
                        email.valid = null;
                    }
                });
                this.showValidationMessage = this.invalidEmails;  //Could this be a computed property?
            }
```